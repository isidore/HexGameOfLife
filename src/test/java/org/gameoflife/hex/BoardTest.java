package org.gameoflife.hex;

import org.approvaltests.Approvals;
import org.approvaltests.namer.NamedEnvironment;
import org.approvaltests.namer.NamerFactory;
import org.junit.jupiter.api.Test;
import org.lambda.query.Query;

import javax.naming.Name;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class BoardTest {
    @Test
    void printEmptyBoard() {
        Approvals.verify(new GameOfLifeBoard());
    }

    @Test
    void printWithCellsAtSpecifiedPosititon() {
        GameOfLifeBoard game = new GameOfLifeBoard(new Cell[]{new Cell(4, 4)});

        Approvals.verify(game);
    }

    @Test
    void testZeroDies() {
        String timeline = "";
        GameOfLifeBoard gameOfLife = new GameOfLifeBoard();
        gameOfLife.setAlive(4, 4);
        timeline += gameOfLife;
        GameOfLifeBoard gameOfLife2 = gameOfLife.advanceTurn();
        timeline += "\n NEXT TURN \n";
        timeline += gameOfLife2;
        Approvals.verify(timeline);
    }

    @Test
    void testTwoFirstLevelNeighbours() {
        String timeline = "";
        GameOfLifeBoard gameOfLife = new GameOfLifeBoard();
        gameOfLife.setAlive(4, 4);
        gameOfLife.setAlive(3, 5);
        gameOfLife.setAlive(4, 6);
        gameOfLife.setAlive(5, 5);
        timeline += gameOfLife;
        GameOfLifeBoard gameOfLife2 = gameOfLife.advanceTurn();
        timeline += "\n NEXT TURN \n";
        timeline += gameOfLife2;
        Approvals.verify(timeline);
    }

    @Test
    void testEverything() {
        Cell cell = new Cell(4, 4);
        List<Cell> level1 = cell.getLevelOneNeighbours();
        List<Cell> level2 = cell.getLevelTwoNeighbours();
        String timeline = "";

        for (int l1 = 0; l1 < level1.size(); l1++) {
            for (int l2 = 0; l2 < level2.size(); l2++) {
                timeline += String.format("\n\nNeighbours(Level 1 , Level 2) = (%s, %s) => \n", l1, l2);
                ArrayList<Cell> cells = new ArrayList<>();
                cells.add(cell);
                cells.addAll(level1.subList(0, l1));
                cells.addAll(level2.subList(0, l2));
                GameOfLifeBoard gameOfLifeBoard = new GameOfLifeBoard(cells);
                timeline += gameOfLifeBoard;
                timeline += "\n NEXT TURN \n";

                timeline += gameOfLifeBoard.advanceTurn();
            }
        }
        Approvals.verify(timeline);
    }
}
