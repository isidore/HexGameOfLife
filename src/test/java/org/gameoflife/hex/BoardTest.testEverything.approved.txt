

Neighbours(Level 1 , Level 2) = (0, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (0, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (0, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (0, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (0, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (0, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   _   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   _   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   _   X   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (1, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   _   X   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (2, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   _   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   X   _   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   X   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   X   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   _   _ 
_   X   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   _   _   
  _   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   _   _ 
_   X   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (3, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   X   X   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   X   _   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   _   _ 
_   X   X   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   X   _   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   X   _ 
_   X   X   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   X   _   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   X   _   X   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  _   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   X   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (4, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 0) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   X   X   _   _ 
_   X   X   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   X   _   _ 
_   X   _   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 1) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   _   _ 
_   X   X   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   _   _ 
_   X   _   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 2) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   X   X   X   _ 
_   X   X   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  _   _   _   X   _ 
_   X   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 3) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 4) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  _   X   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  _   X   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 


Neighbours(Level 1 , Level 2) = (5, 5) => 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   X   X   X   _ 
_   X   X   X   _   
  X   X   _   X   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 

 NEXT TURN 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   X   _   _   
  X   _   _   X   _ 
_   _   _   _   _   
  X   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
_   _   _   _   _   
  _   _   _   _   _ 
